File: 프로그래머스/0/181912. 배열 만들기 5/배열 만들기 5.java
import java.util.ArrayList;

class Solution {
    public int[] solution(String[] intStrs, int k, int s, int l) {
        ArrayList<Integer> list = new ArrayList<>();

        for (String str : intStrs) {
            // 부분 문자열 잘라내기
            String sliced = str.substring(s, s + l);

            // 문자열 → 정수로 변환
            int num = Integer.parseInt(sliced);

            // k보다 큰 값만 추가
            if (num > k) {
                list.add(num);
            }
        }

        // ArrayList → int[] 변환
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }

        return answer;
    }
}

File: 프로그래머스/0/181912. 배열 만들기 5/배열 만들기 5.js
function solution(intStrs, k, s, l) {
    return intStrs
        .map((targetNum) => targetNum.slice(s, s + l))  // 일단 잘라내고
        .map(Number)                                    // 문자열 → 숫자 변환
        .filter((num) => num > k);                      // k보다 큰 것만 남김
}
